api/admin.go

```diff
@@ -17,7 +17,7 @@ type adminUserParams struct {
 	Role         string                 `json:"role"`
 	Email        string                 `json:"email"`
 	Phone        string                 `json:"phone"`
-	Password     string                 `json:"password"`
+	Password     *string                `json:"password"`
```

Comment:
Using a pointer here because when the following request bodies are decoded, the password field in the struct will default to an empty string. This means that we won't be able to tell if the original request intended to modify the password or not. 

```
{ "email": "xxx", "password": "" } 
or 
{ "email": "xxx" }
```

Using a pointer ensures that if the password field is missing, decoding the request body will assign a nil pointer to `password`.